arrow/util.py

```diff
@@ -23,6 +28,45 @@ def is_timestamp(value):
         return False
 
 
+def windows_datetime_from_timestamp(timestamp, tz=None):
+    """Computes datetime from timestamp. Supports negative timestamps on Windows platform."""
+    sec_frac, sec = math.modf(timestamp)
+    dt = datetime.datetime(1970, 1, 1, tzinfo=dateutil.tz.tzutc()) + datetime.timedelta(
+        seconds=sec, microseconds=sec_frac * 1e6
+    )
+    if tz is not None:
+        if tz == dateutil.tz.tzlocal():
```

Comment:
@systemcatch and I just reviewed this PR over a call, and we are a bit confused about the logic here.

We tried the following example:

```shell
>>> import arrow
>>> timestamp = -1572204340.6460679
>>> arrow.util.windows_datetime_from_timestamp(timestamp)
1920-03-07 04:34:19.353932+00:00
```

But when we run `datetime.fromtimestamp` (running on macOS), we get the following:
```shell
>>> from datetime import datetime 
>>> datetime.fromtimestamp(-1572204340.6460679)
datetime.datetime(1920, 3, 6, 23, 34, 19, 353932)
```

I am on the EST timezone (5 hours behind UTC), and it seems that the UTC tzinfo being set on line 34 is persisting when no timezone is passed in. Should we be converting to `tzlocal()` all the time to align with the behavior of `datetime.fromtimestamp` (ref: https://docs.python.org/3.7/library/datetime.html#datetime.datetime.fromtimestamp).

Reply:
In a comment from a few days ago you asked me to change `util.windows_datetime_from_timestamp` to return UTC, as it's more aligned with general arrow behaviour. `util.safe_fromtimestamp` is the one emulating the behaviour  of `datetime.fromtimestamp`

Reply:
Hey @kkoziara that comment I made was probably wrong on reflection, we should revert back to the previous behaviour.

Reply:
Yeah sorry for this @kkoziara, we misspoke. We should maintain the functionality currently present in Arrow 0.15.4, with the addition of Windows negative timestamp support:

```shell
venv ❯ python3
Python 3.7.3 (v3.7.3:ef4ec6ed12, Mar 25 2019, 16:52:21)
[Clang 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import arrow
>>> from datetime import datetime
>>> arrow.__version__
'0.15.4'
>>> timestamp = -1572204340.6460679
>>> arrow.Arrow.fromtimestamp(timestamp)
<Arrow [1920-03-06T23:34:19.353932-05:00]>
>>> arrow.Arrow.utcfromtimestamp(timestamp)
<Arrow [1920-03-07T04:34:19.353932+00:00]>
>>> datetime.fromtimestamp(timestamp)
datetime.datetime(1920, 3, 6, 23, 34, 19, 353932)
>>> datetime.utcfromtimestamp(timestamp)
datetime.datetime(1920, 3, 7, 4, 34, 19, 353932)
```

Also, when parsing timestamps with the "X" token, Arrow seems to explicitly return a UTC timestamp.